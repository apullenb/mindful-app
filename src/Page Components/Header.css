
.header {
  background-color: #0b1f3a;
  color: #b8b6b6;
  font-family: var(--custom-font);
  text-shadow: 0 px 0px 4px 3px rgb(43, 42, 42);
  display: flex; 
  flex-direction: row;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  font-size: 35px;  
  color: white;
  padding-top: 2%;
  padding-bottom: 5%; 
  padding-left: 2%;
  margin-bottom: 1%;
}
/* .Navbar {
  position: fixed;
  z-index: 1;
  top: 0;
  background-color: #01b8d8;
  width:100%
} */
.head{
  margin-right: 25px;
}
.hidden {
  display:none;
}
.nav {
  display: block;
  float: right;
  width: 175px;
  height: 180px;
line-height: 29px;
font-family: sans-serif;
text-decoration: none;
text-align: center;
color: rgb(255, 255, 255);
border: 1px solid rgb(221, 219, 219);
letter-spacing: 2px;
padding-left: 8px;
padding-right:3px;
padding-top:10px;
padding-bottom:5px;
margin: 0px;
background-color: #01b8d8

} 
.show-nav a {
  margin-top: 1px;
  display: block;
  font-size: 18px;
font-family: sans-serif;
text-decoration: none;
  text-decoration: none;
  color: rgb(255, 255, 255);
  padding-left: 12px;
padding-right:12px;
padding-top:2px;
padding-bottom:2px;
transition: all .35s;  
}

.nav-button {
  background-color:  #f7f7f7;
  padding-left:15px;
  padding-right: 15px;
  padding-top:8px;
  padding-bottom:8px;
  font-size: 24px;
  border-radius: 8px;
  border-bottom: 1px solid white;
}

.nav-button:hover {
  background-color: #01b8d8;
  color: #043942;
  padding-left:18px;
  padding-right: 18px;
  padding-top:8px;
  padding-bottom:8px;
}

.nav a:hover {
background-color: #093f5f;
box-shadow: 0px 1px 2px 1px white;
cursor: pointer;
}


.nav a:active {
  background-color: #093f5f;
  padding-left:25px;
  box-shadow: 0px 4px 5px 2px white;
  }
/* .show-nav{
  position: relative;
  z-index: 1;
  width:100px;
  float: right;
  overflow: hidden;
  } */

@media all and (max-width: 800px) {

 

  .nav {
    display: block;
    float: right;
   position: fixed;
    height: 180px;
  line-height: 29px;
  font-family: sans-serif;
  text-decoration: none;
  text-align: center;
  color: rgb(255, 255, 255);
  border: 1px solid rgb(221, 219, 219);
  letter-spacing: 2px;
  padding-left: 8px;
  padding-right:3px;
  padding-top:2px;
  padding-bottom:5px;
  margin-bottom: 8px;
  
  } 
  nav a:hover {
    background-color: #093f5f;
    box-shadow: 0px 1px 2px 1px white;
    cursor: pointer;
    }
  .show-nav a {
    margin-top: 1px;
    display: block;
    font-size: 20px;
  font-family: sans-serif;
  text-decoration: none;
    text-decoration: none;
    color: rgb(255, 255, 255);
     padding-left: 12px; 
  padding-right:12px; 
  padding-top:2px;
  padding-bottom:2px;
  transition: all .35s;  
  margin-bottom: 8px
  }
  .nav-button {
      padding-left:10px;
      padding-right: 10px;
      padding-top:px;
      padding-bottom:6px;
      font-size: 24px;
      border-radius: 8px;
  }
  .nav-button:hover {
    padding-left:10px;
    padding-right: 10px;
    padding-top:6px;
    padding-bottom:8px;
  }
  
  
}

