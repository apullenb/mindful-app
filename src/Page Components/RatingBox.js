import React from 'react';
import moment from 'moment';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';




function RatingBox(props) {
  const elements= props.rate
  console.log('rating box', props)
  const star= <FontAwesomeIcon icon="star" style={{color: 'yellow'}} />
  
  
    
    function bestRating() {
        let rating= 0
        let date = ''
        for (let i=0; i < elements.length; i++) {
       if( (elements[i].rate_happiness + elements[i].rate_focus + elements[i].rate_energy) /3 > rating ) {
            rating = (elements[i].rate_happiness + elements[i].rate_focus + elements[i].rate_energy)/3 
            date = elements[i].date
       }     
    } 
    rating = Math.round(rating) 
    const dateFormat = moment(date).format("LL")
    return <div><h4>Your Best Day Was:</h4><p>{dateFormat}</p>
    <p>Mood Rating: </p>
    {Array(rating).fill(star)}
    </div>
}
    const string = bestRating()
    return (
      <div>
      <span style = {ratingBoxStyle}>
          {string} 
       </span>
      </div>
    );
    }
  

  const ratingBoxStyle = {
    color:'white',
    padding: "5px",
    textAlign: "center",
    maxWidth: "50%",
    border: '2px dashed white',
    // boxShadow: "0px 3px 3px black",
    marginLeft:'30% '  
    
  }
 

  export default RatingBox;