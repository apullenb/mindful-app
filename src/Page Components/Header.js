import React, { useState, useEffect } from "react";
import "./Header.css";
import Navbar from "./Navbar";

function Header(props) {
  return (
    <div className="">
      <section className="">
        {props.isAuth === true ? <Navbar setAuth={props.setAuth} /> : " "}
      </section>
      <section className="header">
        <h1>Mindful</h1>
      </section>
    </div>
  );
}
export default Header;
