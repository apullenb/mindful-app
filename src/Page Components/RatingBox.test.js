import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header'
import renderer from 'react-test-renderer';
import RatingBox from './RatingBox';

it('renders without crashing', () => {

  const div = document.createElement('div');

  ReactDOM.render(<RatingBox />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<RatingBox />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });