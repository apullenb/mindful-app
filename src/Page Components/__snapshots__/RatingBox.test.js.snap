// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="header"
>
  <section
    className=""
  >
     
  </section>
  <h1>
    Mindful
  </h1>
</div>
`;
